plenum/server/future_primaries.py

```diff
@@ -26,52 +24,37 @@ def __init__(self, node, node_names, ordered_node_ids, current_required):
-                    and i not in three_pc_batch.invalid_indices:
-
-                node_nym = req.operation.get(TARGET_NYM)
-                node_name = req.operation.get(DATA).get(ALIAS)
-                curName = self.future_ordered_node_ids.get(node_nym)
-                if curName is None:
-                    self.future_ordered_node_ids[node_nym] = node_name
-                elif curName != node_name:
-                    raise LogicError("Alias inconsistency")
-
-                serv = req.operation.get(DATA).get(SERVICES)
-                if VALIDATOR in serv and node_name not in self.future_node_names:
-                    self.future_node_names.append(node_name)
-                elif serv == [] and node_name in self.future_node_names:
-                    self.future_node_names.remove(node_name)
-
-                count = self.count_required_number_of_instances()
-                if self.future_required_number_of_instances != \
-                        count:
-                    self.future_required_number_of_instances = count
-                    self.reselect()
-        return self.future_primaries
-
-    def reselect(self):
-        self.future_primaries = []
-        # Logic similar to select_primaries in node.py
-        master_primary_name, _ = self.node.elector.next_primary_replica_name_for_master(
-            self.future_node_names, self.future_ordered_node_ids)
-        self.future_primaries.append(master_primary_name)
-        primary_rank = self.node.get_rank_by_name(master_primary_name)
-
-        for i in range(1, self.future_required_number_of_instances):
-            new_primary_name, _ = \
-                self.node.elector.next_primary_replica_name_for_backup(
-                    i, primary_rank, self.future_primaries)
+        self.primaries = []
+
+    def handle_pool_request(self, request: Request):
+        if request.operation.get(TXN_TYPE) == NODE \
+                and request.operation.get(DATA).get(SERVICES) is not None:
+
+            node_nym = request.operation.get(TARGET_NYM)
+            node_name = request.operation.get(DATA).get(ALIAS)
+            curName = self.future_ordered_node_ids.get(node_nym)
+            if curName is None:
+                self.future_ordered_node_ids[node_nym] = node_name
+            elif curName != node_name:
+                raise LogicError("Alias inconsistency")
+
+            serv = request.operation.get(DATA).get(SERVICES)
+            if VALIDATOR in serv and node_name not in self.future_node_reg:
+                self.future_node_reg.append(node_name)
+            elif serv == [] and node_name in self.future_node_reg:
+                self.future_node_reg.remove(node_name)
+
+            count = self.count_required_number_of_instances()
+            if self.future_required_number_of_instances != count:
+                self.future_required_number_of_instances = count
+                self.primaries = list(self.node.elector.process_selection(
```

Comment:
I don't like that this class is still not stateless. We have to take are about revert logic, and so on.
Why don't we just have one static method in this class:
```
@staticmethod
def calculate_primaries(self, requests: List[Request], node_reg, ordered_node_ids) -> Optional[List[str]]: # returns new primaries or None if they are not changed
```

Reply:
ok